pkg/pipeline/transform/kubernetes/cni/ovn_kubernetes.go

```diff
@@ -43,27 +43,49 @@ func AddOvnIPs(ips []string, node *v1.Node) []string {
 	return ips
 }
 
+func unmarshalAnnotation(annot []byte) (string, error) {
+	// Depending on OVN (OCP) version, the annotation might be JSON-encoded as a string, or an array of strings
+	// Try first as string (legacy)
+	var subnetsAsString map[string]string
+	err := json.Unmarshal(annot, &subnetsAsString)
+	if err == nil {
+		if subnet, ok := subnetsAsString["default"]; ok {
+			return subnet, nil
+		}
+		return "", fmt.Errorf("unexpected content for annotation %s: %s", ovnSubnetAnnotation, annot)
```

Comment:
(Minor) Wouldn't it be better to try the new way first and then if it fails, try legacy?